engine/src/main/java/org/terasology/engine/rendering/nui/skin/UISkinWithUrn.java

```diff
@@ -0,0 +1,33 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.rendering.nui.skin;
+
+import org.terasology.assets.ResourceUrn;
+import org.terasology.nui.skin.UISkin;
+import org.terasology.nui.skin.UIStyleFamily;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class UISkinWithUrn extends UISkin {
```

Comment:
Let's add this as a remark here for future reference.

```suggestion
/**
 * [NOTE] Sub-classing UISkin into UISkinWithUrn is a work-around in order to associate a ResourceUrn with a UISkin.
 * When a UISkin is deserialised, then it is copied into a new UISkinWithUrn instance to associate it with the urn it was deserialised with.
 *
 * (2021-03-31) It is not clear whether this feature is used anywhere (in the Omega module space) right now.
 */
public class UISkinWithUrn extends UISkin {
```

Reply:
I've added the suggested comment, also clarifying which that this refers to deserialisation from `UISkinTypeHandler` (the UISkin is deserialised from JSON by `UISkinFormat` into data for a `UISkinAsset` and the `UISkinTypeHandler` fetches the skin from its asset).